src/Services/RabbitMQService.cs

```diff
@@ -10,15 +10,50 @@ internal sealed class RabbitMQService : IRabbitMQService
     {
         private IModel _channel;
         private IBasicPublishBatch _batch;
+        private string _connectionString;
         private string _hostName;
         private string _queueName;
+        private string _userName;
+        private string _password;
+        private int _port;
 
-        public RabbitMQService(string hostName, string queueName)
+        public RabbitMQService(string connectionString, string hostName, string queueName, string userName, string password, int port)
         {
-            _hostName = hostName ?? throw new ArgumentNullException(nameof(hostName));
+            _connectionString = connectionString;
+            _hostName = hostName;
             _queueName = queueName ?? throw new ArgumentNullException(nameof(queueName));
+            _userName = userName;
+            _password = password;
+            _port = port;
+
+            ConnectionFactory connectionFactory = new ConnectionFactory();
+
+            // Only set these if specified by user. Otherwise, API will use default parameters.
+            if (!string.IsNullOrEmpty(_connectionString))
+            {
+                connectionFactory.Uri = new Uri(_connectionString);
+            }
+
+            if (!string.IsNullOrEmpty(_hostName))
+            {
+                connectionFactory.HostName = _hostName;
+            }
+
+            if (!string.IsNullOrEmpty(_userName))
+            {
+                connectionFactory.UserName = _userName;
+            }
+
+            if (!string.IsNullOrEmpty(_password))
+            {
+                connectionFactory.Password = _password;
+            }
+
+            if (_port != 0)
+            {
+                connectionFactory.Port = _port;
+            }
 
-            ConnectionFactory connectionFactory = new ConnectionFactory() { HostName = _hostName };
             _channel = connectionFactory.CreateConnection().CreateModel();

```

Comment:
Nit: Instead of exposing GetBatch, GetChannel methods, expose getters
```
 public IBasicPublishBatch Batch => _batch;

public IModel Channel => _channel;
```
Change IRabbitMQService to
```
   IModel Channel { get; }

   IBasicPublishBatch Batch { get; }
```

Reply:
Also, rename channel to  Model as the description here: https://www.rabbitmq.com/releases/rabbitmq-dotnet-client/v3.4.2/rabbitmq-dotnet-client-3.4.2-client-htmldoc/html/type-RabbitMQ.Client.IModel.html states IModel is interface for AMQP Model

Reply:
Rename Batch to BasicPublishBatch. Although BasicPublishBatch is not descriptive, name Batch is misleading